apphub/image_classification/pyramidnet/pyramidnet_torch.py

```diff
@@ -0,0 +1,308 @@
+                    blocks[depth] = Bottleneck
+                    temp_cfg = int((depth - 2) / 12)
+                else:
+                    blocks[depth] = BasicBlock
+                    temp_cfg = int((depth - 2) / 8)
+
+                layers[depth] = [temp_cfg, temp_cfg, temp_cfg, temp_cfg]
+                print('=> the layer configuration for each stage is set to', layers[depth])
+
+            self.inplanes = 64
+            self.addrate = alpha / (sum(layers[depth]) * 1.0)
+
+            self.input_featuremap_dim = self.inplanes
+            self.conv1 = nn.Conv2d(3, self.input_featuremap_dim, kernel_size=7, stride=2, padding=3, bias=False)
+            self.bn1 = nn.BatchNorm2d(self.input_featuremap_dim)
+            self.relu = nn.ReLU(inplace=True)
+            self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)
+
+            self.featuremap_dim = self.input_featuremap_dim
+            self.layer1 = self.pyramidal_make_layer(blocks[depth], layers[depth][0])
+            self.layer2 = self.pyramidal_make_layer(blocks[depth], layers[depth][1], stride=2)
+            self.layer3 = self.pyramidal_make_layer(blocks[depth], layers[depth][2], stride=2)
+            self.layer4 = self.pyramidal_make_layer(blocks[depth], layers[depth][3], stride=2)
+
+            self.final_featuremap_dim = self.input_featuremap_dim
+            self.bn_final = nn.BatchNorm2d(self.final_featuremap_dim)
+            self.relu_final = nn.ReLU(inplace=True)
+            self.avgpool = nn.AvgPool2d(7)
+            self.fc = nn.Linear(self.final_featuremap_dim, num_classes)
+
+        for m in self.modules():
+            if isinstance(m, nn.Conv2d):
+                n = m.kernel_size[0] * m.kernel_size[1] * m.out_channels
+                m.weight.data.normal_(0, math.sqrt(2. / n))
+            elif isinstance(m, nn.BatchNorm2d):
+                m.weight.data.fill_(1)
+                m.bias.data.zero_()
+
+    def pyramidal_make_layer(self, block, block_depth, stride=1):
+        downsample = None
+        if stride != 1:  # or self.inplanes != int(round(featuremap_dim_1st)) * block.outchannel_ratio:
+            downsample = nn.AvgPool2d((2, 2), stride=(2, 2), ceil_mode=True)
+
+        layers = []
+        self.featuremap_dim = self.featuremap_dim + self.addrate
+        layers.append(block(self.input_featuremap_dim, int(round(self.featuremap_dim)), stride, downsample))
+        for i in range(1, block_depth):
+            temp_featuremap_dim = self.featuremap_dim + self.addrate
+            layers.append(
+                block(int(round(self.featuremap_dim)) * block.outchannel_ratio, int(round(temp_featuremap_dim)), 1))
+            self.featuremap_dim = temp_featuremap_dim
+        self.input_featuremap_dim = int(round(self.featuremap_dim)) * block.outchannel_ratio
+
+        return nn.Sequential(*layers)
+
+    def forward(self, x):
+        if self.dataset == 'cifar10' or self.dataset == 'cifar100':
```

Comment:
why is this architecture hard-coded to specific datasets?

Reply:
that's done by paper authors.